app/javascript/client/product/modals/ProductModal.tsx

```diff
@@ -1,48 +1,108 @@
-import React, { FC, useContext } from 'react';
+import React, { FC, useContext, useState, useRef, useEffect, FocusEvent } from 'react';
+import { ProductInfoFragment, useAddToCartMutation } from '../../graphqlTypes';
+import { Field, Form, Formik, FormikHelpers } from "formik";
+import { FormikNumberInput, FormikTextInput} from '../../form/inputs';
 import styled from 'styled-components';
 import { ModalContext } from '../../home/HomePage';
-import CSS from 'csstype';
 
-const ProductModalContainer = styled.div`
+const AddToCartButton = styled.button`
+
+`
+
+const CloseModalButton = styled.button`
+    width: 20px;
+    height: 20px;
+    cursor: pointer;
+    background: none;
+    border: 1px solid black;
+    border-radius: 1rem;
+    font-weight: bold;
+`
+
+const ProductInformation = styled.div`
     display: flex;
     flex-direction: column;
-    justify-content: center;
+`
+
+const ProductImageContainer = styled.img`
+    object-fit: cover;
+    width: 290px;
+    height: 390px;
+`;
+
+const ProductModalContainer = styled.div`
+    display: grid;
     align-items: center;
     background: white;
     border-radius: 10px;
     padding: 10px;
 `;
 
-const CloseModalButton = styled.button`
-    width: 100px;
-    height: 50px;
-    cursor: pointer;
+const ProductName = styled.div`
+    font-size: 18px;
+    font-weight: bold;
 `
+
 interface ProductModalProps {}
 
+interface AddToCartData {
+    productId: string;
+    quantity: number;
+}
+
 const ProductModal: FC<ProductModalProps> = () => {
 
     const { selectedProduct, closeModal} = useContext(ModalContext);
+    
+    const inputRef: React.RefObject<HTMLInputElement> = useRef(null);
 
-    if (!selectedProduct.imageUrl) return null;
+    const [addItemToCart] = useAddToCartMutation();   
+
+    const handleSubmit = async (values: AddToCartData, formikHelpers: FormikHelpers<AddToCartData>) => {
+        if (!inputRef.current) return initialValues;
+
+        const productQuantity = parseInt(inputRef.current.value);
+        
+        const addedItem = addItemToCart({
+            variables: {
+                input: {
+                    productId: selectedProduct.id,
+                    quantity: productQuantity
+                }
+            }
+        });
 
-    const imageStyles: CSS.Properties = {
-        objectFit: 'cover',
-        width: '300px',
-        height: '400px'
-    }
+        console.log("added " + productQuantity + " to cart");
+        console.log(addedItem);
+        closeModal();
```

Comment:
I wonder if we should try to add a flash-type element to let the person know that the items have successfully been added to the cart ?? Maybe with a Context Provider and we pass in a message to flash, and have a component on the home page that shows it, something like that ?? (For a future PR, of course)